data_structures/src/superblock.rs

```diff
@@ -265,31 +254,50 @@ impl SuperBlockState {
 }
 
 /// Produces a `SuperBlock` that includes the blocks in `block_headers` if there is at least one of them.
+/// // remove return Option
 pub fn mining_build_superblock(
     block_headers: &[BlockHeader],
     ars_ordered_hash_leaves: &[Hash],
     index: u32,
     last_block_in_previous_superblock: Hash,
-) -> Option<SuperBlock> {
-    let last_block = block_headers.last()?.hash();
-    let merkle_drs: Vec<Hash> = block_headers
-        .iter()
-        .map(|b| b.merkle_roots.dr_hash_merkle_root)
-        .collect();
-    let merkle_tallies: Vec<Hash> = block_headers
-        .iter()
-        .map(|b| b.merkle_roots.tally_hash_merkle_root)
-        .collect();
-
-    Some(SuperBlock {
-        ars_length: ars_ordered_hash_leaves.len() as u64,
-        data_request_root: hash_merkle_tree_root(&merkle_drs),
-        tally_root: hash_merkle_tree_root(&merkle_tallies),
-        ars_root: hash_merkle_tree_root(ars_ordered_hash_leaves),
-        index,
-        last_block,
-        last_block_in_previous_superblock,
-    })
+) -> SuperBlock {
+    let last_block = block_headers.last();
+    match last_block {
+        None =>
+        // Build "empty" Superblock (there were no blocks during super-epoch)
+        {
+            SuperBlock {
+                ars_length: ars_ordered_hash_leaves.len() as u64,
+                ars_root: hash_merkle_tree_root(ars_ordered_hash_leaves),
+                data_request_root: Hash::default(),
+                tally_root: Hash::default(),
+                index,
+                last_block: last_block_in_previous_superblock,
+                last_block_in_previous_superblock,
+            }
```

Comment:
I sort of like this better just to keep focus on the fields that are not being defaulted and are relevant to each case:
```suggestion
            SuperBlock {
                ars_length: ars_ordered_hash_leaves.len() as u64,
                ars_root: hash_merkle_tree_root(ars_ordered_hash_leaves),
                index,
                last_block: last_block_in_previous_superblock,
                last_block_in_previous_superblock,
                ..Default::default()
            }
```

Note however that this is a code style preference, I have no real problem with how it is right now :sweat_smile: